packages/dev/core/src/Cameras/Inputs/BaseCameraPointersInput.ts

```diff
@@ -69,40 +75,38 @@ export abstract class BaseCameraPointersInput implements ICameraInput<Camera> {
         this._shiftKey = false;
         this._buttonsPressed = 0;
 
-        this._connectedObserver = this.camera._deviceSourceManager!.onDeviceConnectedObservable.add((deviceSource) => {
+        this._connectedObserver = deviceSourceManager.onDeviceConnectedObservable.add((deviceSource) => {
             if (deviceSource.deviceType === DeviceType.Mouse) {
                 this._mouseObserver = deviceSource.onInputChangedObservable.add((eventData) => {
-                    if (!("deltaY" in eventData)) {
-                        let type = PointerEventTypes.POINTERMOVE;
-
-                        if (eventData.inputIndex !== PointerInput.Move) {
-                            type = deviceSource.getInput(eventData.inputIndex) === 1 ? PointerEventTypes.POINTERDOWN : PointerEventTypes.POINTERUP;
-                        }
-
-                        this._pointerInput(eventData, type);
-
-                        if (Date.now() - this._previousStartingPointerTime < InputManager.DoubleClickDelay && !this._doubleClickOccured && type === PointerEventTypes.POINTERUP) {
-                            this._doubleClickOccured = true;
-                            type = PointerEventTypes.POINTERDOUBLETAP;
-                            this._pointerInput(eventData, type);
-                        } else {
-                            this._doubleClickOccured = false;
-                        }
-
-                        if (type === PointerEventTypes.POINTERUP || type === PointerEventTypes.POINTERDOUBLETAP) {
-                            this._previousStartingPointerTime = Date.now();
-                        }
+                    const type = GestureRecognizer.DeterminePointerEventType(deviceSource, eventData);
+                    const pointerEventData = eventData as IPointerEvent;
+
+                    if ((type & PointerEventTypes.POINTERMOVE) !== 0) {
+                        this._pointerInput(pointerEventData, PointerEventTypes.POINTERMOVE);
+                    }
+                    if ((type & PointerEventTypes.POINTERDOWN) !== 0) {
+                        this._pointerInput(pointerEventData, PointerEventTypes.POINTERDOWN);
+                    }
+                    if ((type & PointerEventTypes.POINTERUP) !== 0) {
+                        this._pointerInput(pointerEventData, PointerEventTypes.POINTERUP);
+                    }
+                    if ((type & PointerEventTypes.POINTERDOUBLETAP) !== 0) {
+                        this._pointerInput(pointerEventData, PointerEventTypes.POINTERDOUBLETAP);
                     }

```

Comment:
Most of these are mutually exclusive. Maybe this?
```suggestion
                    if ((type & PointerEventTypes.POINTERMOVE) !== 0) {
                        this._pointerInput(pointerEventData, PointerEventTypes.POINTERMOVE);
                    } else if ((type & PointerEventTypes.POINTERDOWN) !== 0) {
                        this._pointerInput(pointerEventData, PointerEventTypes.POINTERDOWN);
                    } else if ((type & PointerEventTypes.POINTERUP) !== 0) {
                        this._pointerInput(pointerEventData, PointerEventTypes.POINTERUP);

                        if ((type & PointerEventTypes.POINTERDOUBLETAP) !== 0) {
                            this._pointerInput(pointerEventData, PointerEventTypes.POINTERDOUBLETAP);
                        }
                    }
```